week2/react-exercises/dog/src/App.js

```diff
@@ -0,0 +1,57 @@
+import React, {useState, useEffect} from 'react'
+import './App.css'
+
+const Button = ({getDogPhoto}) => {
+  return (
+    <button onClick={getDogPhoto}>Get a dog!</button>
+  )
+}
+
+const DogPhoto = ({dogPhoto}) => {
+  return (
+    <>
+      <img className="c-image" src={dogPhoto} alt="Dog"></img>
+    </>
+  )
+}
+
+const API_URL ='https://dog.ceo/api/breeds/image/random';
+
+const DogGallery = () => {
+
+  const [dogPhotos, setDogPhotos] = useState([]);
+
+  const getDogPhoto = async () => {
```

Comment:
It is better for these functions to live inside the useEffect. Because in every rerender of the component it will be defined again. If it lives inside the useEffect it will be defined only once

Reply:
Can you be more specific? I get this error

> React Hook "useState" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function

Reply:
The way I propose getDogPhoto is only declared once and not for each rerender
```suggestion
  useEffect(() => {
     const getDogPhoto = async () => {...}
     getDogPhoto();
  }, [])
```

Reply:
Ok.. Now I get it.